go/pkg/casng/batching.go

```diff
@@ -665,3 +657,27 @@ func (u *BatchingUploader) UploadTree(ctx context.Context, execRoot impath.Absol
 
 	return
 }
+
+// ReplaceWorkingDir swaps remoteWorkingDir for workingDir in path which must be prefixed by root.
+// workingDir is assumed to be prefixed by root, and the returned path will be a descendent of root, but necessarily a decendent of remoteWorkingDir.
+// Example: path=/root/out/foo.c, root=/root, workdingDir=out/reclient, remoteWorkingDir=set_by_reclient/a/b, result=/root/set_by_reclient/a/foo.c
```

Comment:
Is that fact that  remoteWorkingDir=`set_by_reclient/a/b` but result `/root/set_by_reclient/a/foo.c` (without `/b`) on purpose or is it a typo?

Reply:
On purpose. The unit test has a case for this one. The path resolves to `../foo.c` relative to `out/reclient`, which is then joined to `set_by_reclient_/a/b/../foo.c` which normalizes to `set_by_reclient/a/foo.c`.